testProjectKotlinDslBase/build.gradle.kts

```diff
@@ -100,7 +100,7 @@ tasks {
             from(sourceSet.output)
 
             val compileTaskName = sourceSet.getCompileTaskName("java")
-            dependsOn(tasks.getByName(compileTaskName))
+            dependsOn(project.tasks.getByName(compileTaskName))
```

Comment:
An additional extension was added to the task container in 1.0.3. So in order to resolve ambiguity in method resolution I had to explicitly qualify the ```tasks``` property.